src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,25 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> queue = new CopyOnWriteArrayList<>();
```

Comment:
Using CopyOnWriteArrayList ensures thread-safety for concurrent modifications, but it might not be the most efficient choice if there are frequent writes, as this data structure creates a new copy of the underlying array on each modification. If writes are less frequent than reads and iteration, this is fine. Otherwise, consider using a different concurrent collection that better suits your use case.